pkg/counter/counter.go

```diff
@@ -1,96 +1,84 @@
 package counter
 
 import (
+	"errors"
 	"fmt"
+	"math"
 	"strconv"
 	"sync/atomic"
-
-	"github.com/aquasecurity/tracee/pkg/errfmt"
 )
 
 type Counter struct {
-	count uint64
+	value uint64
 }
 
-func NewCounter(v uint64) Counter {
-	return Counter{
-		count: v,
-	}
+// NewCounter creates a new counter with the given initial value.
+func NewCounter(initialValue uint64) Counter {
+	return Counter{value: initialValue}
 }
 
-func (c *Counter) Increment(amount ...uint64) error {
-	if len(amount) == 0 {
-		return c.incAtomic(1)
-	}
-
-	sum, err := sumUint64(amount...)
-	if err != nil {
-		return errfmt.WrapError(err)
-	}
-
-	return c.incAtomic(sum)
+// Increase increases the counter by the given value (thread-safe).
+func (c *Counter) Increase(values ...uint64) error {
+	_, err := c.IncreaseAndRead(values...)
+	return err
 }
 
-func (c *Counter) Decrement(amount ...uint64) error {
-	if len(amount) == 0 {
-		return c.decAtomic(1)
+// IncreaseAndRead increases the counter by the given value (thread-safe) and returns the new value.
+func (c *Counter) IncreaseAndRead(values ...uint64) (uint64, error) {
```

Comment:
I think increment/decrement was a better naming. Not sure why you say increase/decrease is more widely used - if you google for "increase counter" you will get most results which say that the operation is called increment/decrement and it causes the value of the counter to be increased/decreased

Reply:
Not against a re-re-naming.